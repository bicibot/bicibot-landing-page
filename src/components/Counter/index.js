import React from "react";
import styled from "styled-components";

const NumberContainer = styled.div`
  background: #ffffff;
  box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.12);
  border-radius: 4px;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 20px;
  flex-direction: column;
  margin-right: 15px;
  color: #5a31f8;
  font-size: 64px;
  font-weight: bold;
`;

const CounterContainer = styled.div`
  display: flex;
  flex-direction: row;
`;

const ReportsContainer = styled.div`
  margin: 20px;
  h2 {
    margin-top: 10px;
    color: #5a31f8;
    font-size: 36px;
  }
`;

export default function Counter(props) {
  const numArray = [...`${props.total}`].map(n => +n);
  const reportsCounter = numArray.map((num, i) => (
    <NumberContainer key={i}>{num}</NumberContainer>
  ));

  return (
    <>
      <ReportsContainer>
        <CounterContainer>{reportsCounter}</CounterContainer>
        <h2>denÃºncias</h2>
      </ReportsContainer>
    </>
  );
}
