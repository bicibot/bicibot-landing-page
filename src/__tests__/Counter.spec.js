import React from "react";
import { create } from "react-test-renderer";
import Counter from "../components/Counter"

describe("Counter component", () => {
    test("Matches the snapshot", () => {
        const counter = create(<Counter total={402} />);
        expect(counter.toJSON()).toMatchSnapshot();
    });
});