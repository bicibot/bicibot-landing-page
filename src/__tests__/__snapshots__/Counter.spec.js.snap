// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component Matches the snapshot 1`] = `
<div
  className="sc-htpNat dFcVJV"
>
  <div
    className="sc-bwzfXH fpBhud"
  >
    <div
      className="sc-bdVaJa qkxPG"
    >
      4
    </div>
    <div
      className="sc-bdVaJa qkxPG"
    >
      0
    </div>
    <div
      className="sc-bdVaJa qkxPG"
    >
      2
    </div>
  </div>
  <h2>
    den√∫ncias
  </h2>
</div>
`;
